src/code/z_camera.c

```diff
@@ -526,282 +869,7416 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
+    curUID = sNextUID;
+    sNextUID++;
+    while (curUID != 0) {
+        if (curUID == 0) {
+            sNextUID++;
+        }
+
+        for (j = 0; j < NUM_CAMS; j++) {
+            cameraPtr = camera->play->cameraPtrs[j];
+            if ((cameraPtr != NULL) && (curUID == cameraPtr->uid)) {
+                break;
+            }
+        }
+
+        if (j == NUM_CAMS) {
+            break;
+        }
+
+        curUID = sNextUID++;
+    }
+
+    camera->inputDir.y = 0x4000;
+    camera->uid = curUID;
+    camera->camDir = camera->inputDir;
+    camera->nextCamSceneDataId = -1;
+    camera->up.z = camera->up.x = 0.0f;
+    camera->up.y = 1.0f;
+    camera->fov = 60.0f;
+    camera->yOffsetUpdateRate = 0.05f;
+    camera->xzOffsetUpdateRate = 0.05f;
+    camera->fovUpdateRate = 0.05f;
+    camera->rUpdateRateInv = 10.0f;
+    camera->yawUpdateRateInv = 10.0f;
+    camera->pitchUpdateRateInv = 16.0f;
+
+    sCameraLetterboxSize = 32;
+    sCameraHudVisibility = 0;
+
+    camera->setting = camera->prevSetting = CAM_SET_FREE0;
+    camera->bgCamIndex = camera->prevBgCamDataId = -1;
+    camera->stateFlags = 0;
+    camera->mode = CAM_MODE_NORMAL;
+    camera->bgId = BGCHECK_SCENE;
+    camera->unk168 = 0xF;
+    camera->timer = -1;
+    camera->focalActor = NULL;
+    camera->target = NULL;
+    Camera_SetStateFlag(camera, CAM_STATE_INITIALIZED);
+    camera->quakeOffset.z = camera->quakeOffset.y = camera->quakeOffset.x = 0;
+    camera->up.z = camera->up.x = 0.0f;
+    camera->atLerpStepScale = 1;
+    camera->up.y = 1.0f;
+    sCameraInterfaceFlags = CAM_INTERFACE_FLAGS(CAM_LETTERBOX_IGNORE, CAM_HUD_VISIBILITY_IGNORE, 0);
+    sCameraInitSceneTimer = 3;
+}
+
+/**
+ * OoT: func_80057FC4
+ */
+void func_800DDFE0(Camera* camera) {
+    if (camera != &camera->play->mainCamera) {
+        camera->prevSetting = camera->setting = CAM_SET_FREE0;
+        Camera_UnsetStateFlag(camera, CAM_STATE_2);
+    } else {
+        switch (camera->play->roomCtx.currRoom.unk3) {
+            case 1:
+                camera->prevSetting = CAM_SET_DUNGEON0;
+                Camera_ChangeSettingFlags(camera, CAM_SET_DUNGEON0, CAM_CHANGE_SETTING_1);
+                break;
+
+            case 0:
+                camera->prevSetting = CAM_SET_NORMAL0;
+                Camera_ChangeSettingFlags(camera, CAM_SET_NORMAL0, CAM_CHANGE_SETTING_1);
+                break;
+
+            case 2:
+                camera->prevSetting = CAM_SET_ROOM0;
+                Camera_ChangeSettingFlags(camera, CAM_SET_ROOM0, CAM_CHANGE_SETTING_1);
+                break;
+
+            default:
+                camera->prevSetting = CAM_SET_NORMAL0;
+                Camera_ChangeSettingFlags(camera, CAM_SET_NORMAL0, CAM_CHANGE_SETTING_1);
+                break;
+        }
+        Camera_SetStateFlag(camera, CAM_STATE_2);
+    }
+}
+
+/**
+ * Unused empty function
+ */
+void Camera_Stub800DE0E0(Camera* camera) {
+}
+
+void Camera_InitPlayerSettings(Camera* camera, Player* player) {
```

Comment:
I don't see anything player about this, and the rest of the system was genericized into an actor. I don't see why this func couldn't match with actor.